admin-dev/themes/new-theme/js/pages/product/edit/helpers/product-form.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import BigNumber from 'bignumber.js';
+
+export const calculateTax = (taxRate: string): BigNumber => {
+  let newTaxRate;
+
+  try {
+    newTaxRate = new BigNumber(taxRate);
+  } catch (error) {
+    newTaxRate = new BigNumber(NaN);
+  }
+
+  if (newTaxRate.isNaN()) {
+    newTaxRate = new BigNumber(0);
```

Comment:
Normally you would throw an exception here

Reply:
The exception is already thrown by BigNumber tbh and the logic should probably be handler by the parent component if the tax ratio is undefined don't you think?